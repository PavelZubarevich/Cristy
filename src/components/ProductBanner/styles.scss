@import '../../styles/vars.scss';

.productBanner {
  background-color: #eef1f8;
  padding: 135px 0;

  @media(max-width: 768px) {
    padding: 60px 0;
  }

  @media(max-width: 576px) {
    padding: 30px 0;
  }

  &__body {
    display: flex;
    justify-content: space-between;
    align-items: center;
    gap: 50px;

    &.reversed {
      flex-direction: row-reverse;
    }
  }

  &__content {
    flex: 1;
  }

  &__title {
    font-family: 'Butler';
    font-size: 65px;
    font-weight: 700;
    color: $primary;
    margin-bottom: 20px;
    text-transform: capitalize;

    @media(max-width: 768px) {
      font-size: 50px;
    }

    @media(max-width: 576px) {
      font-size: 35px;
      margin-bottom: 10px;
    }
  }

  &__description {
    font-family: 'Raleway';
    font-size: 14px;
    font-weight: 400;
    color: $primaryText;
    margin-bottom: 20px;
  }

  &__link {
    font-family: 'Lora';
    font-style: italic;
    font-size: 15px;
    font-weight: 400;
    text-transform: capitalize;
    color: $primary;
    display: flex;
    align-items: center;
    gap: 10px;

    &>svg {
      font-size: 19px;
    }
  }

  &__image {
    flex: 1;
    height: 350px;

    @media(max-width: 768px) {
      display: none;
    }

    &>img {
      width: 100%;
      height: 100%;
      object-fit: contain;
    }
  }
}