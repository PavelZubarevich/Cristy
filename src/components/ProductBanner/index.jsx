import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';
import Button from '../common/Button';
import './styles.scss';

function ProductBanner({
  title, description, image, route, reversed,
}) {
  const navigate = useNavigate();

  const handleButtonClick = () => {
    navigate(route);
  };

  return (
    <div className="productBanner">
      <div className="container">
        <div className={`productBanner__body ${reversed ? 'reversed' : ''}`}>
          <div className="productBanner__content">
            <h2 className="productBanner__title">{title}</h2>
            <p className="productBanner__description">{description}</p>
            <Button variant="contained" color="primary" onClick={handleButtonClick}>
              Buy now
            </Button>
          </div>
          <div className="productBanner__image">
            <div className="productBanner__image">
              <img src={image} alt={title} />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default ProductBanner;

ProductBanner.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  image: PropTypes.string,
  route: PropTypes.string,
  reversed: PropTypes.bool,
};

ProductBanner.defaultProps = {
  title: '',
  description: '',
  image: '',
  route: '',
  reversed: false,
};
