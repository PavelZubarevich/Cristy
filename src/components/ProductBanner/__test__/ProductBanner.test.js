import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import ProductBanner from '..';

const mockedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  useNavigate: () => mockedNavigate,
}));

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (str) => str,
    i18n: {
      changeLanguage: () => new Promise(() => { }),
      options: {
        supportedLngs: ['ru', 'en', 'cimode'],
      },
    },
  }),
}));

describe('ProductBanner', () => {
  it('should render product banner', () => {
    render(
      <ProductBanner
        title="title"
        description="description"
        image=""
        route="testroute/1"
      />,
      { wrapper: BrowserRouter },
    );

    expect(document.body).toMatchSnapshot();
  });

  it('should render reversed product banner', () => {
    render(
      <ProductBanner
        title="title"
        description="description"
        image=""
        route="testroute/1"
        reversed
      />,
      { wrapper: BrowserRouter },
    );

    expect(document.body).toMatchSnapshot();
  });

  it('should handle navagate to product page', () => {
    render(
      <ProductBanner
        title="title"
        description="description"
        image=""
        route="testroute/1"
        reversed
      />,
      { wrapper: BrowserRouter },
    );

    userEvent.click(screen.getByRole('button', { text: /BuyNowLabel/i }));

    expect(mockedNavigate).toHaveBeenCalledTimes(1);
    expect(mockedNavigate).toHaveBeenCalledWith('testroute/1');
  });
});
