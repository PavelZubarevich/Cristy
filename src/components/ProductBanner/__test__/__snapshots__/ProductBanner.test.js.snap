// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductBanner should render product banner 1`] = `
<body>
  <div>
    <div
      class="productBanner"
    >
      <div
        class="container"
      >
        <div
          class="productBanner__body "
        >
          <div
            class="productBanner__content"
          >
            <h2
              class="productBanner__title"
            >
              title
            </h2>
            <p
              class="productBanner__description"
            >
              description
            </p>
            <button
              class="button button_lg button_contained button_contained_primary"
              type="button"
            >
              BuyNowLabel
            </button>
          </div>
          <div
            class="productBanner__image"
          >
            <div
              class="productBanner__image"
            >
              <img
                alt="title"
                src=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ProductBanner should render reversed product banner 1`] = `
<body>
  <div>
    <div
      class="productBanner"
    >
      <div
        class="container"
      >
        <div
          class="productBanner__body reversed"
        >
          <div
            class="productBanner__content"
          >
            <h2
              class="productBanner__title"
            >
              title
            </h2>
            <p
              class="productBanner__description"
            >
              description
            </p>
            <button
              class="button button_lg button_contained button_contained_primary"
              type="button"
            >
              BuyNowLabel
            </button>
          </div>
          <div
            class="productBanner__image"
          >
            <div
              class="productBanner__image"
            >
              <img
                alt="title"
                src=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
